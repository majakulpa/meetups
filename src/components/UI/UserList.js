import React from 'react'
import { Link } from 'react-router-dom'

const userList = ({ usersArr, user, text }) => {
  return (
    <React.Fragment>
      {usersArr.length > 0 && (
        <React.Fragment>
          <span className="font-bold text-lg">
            {text} ({usersArr.length})
          </span>
          <div className="flex flex-wrap justify-center lg:justify-start">
            {usersArr.map(attendee => (
              <Link
                className="flex flex-col items-center w-32 md:w-48 bg-gray-100 rounded 
                p-2 md:p-3 m-1 md:m-3 hover:bg-gray-200 hover:shadow"
                key={attendee.id}
                to={
                  user && user.id === attendee.id
                    ? `/my-account/${attendee.id}`
                    : `/users/${attendee.id}`
                }
              >
                <div
                  className="h-16 w-16 bg-cover rounded-full bg-center"
                  style={{
                    backgroundImage: `url(${attendee.profileImage}})`
                  }}
                  title="Profile Image"
                ></div>
                <div className="text-sm font-medium m-3">{attendee.name}</div>
              </Link>
            ))}
          </div>
        </React.Fragment>
      )}
    </React.Fragment>
  )
}

export default userList
