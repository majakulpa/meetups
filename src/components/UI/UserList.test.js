import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import { cleanup } from '@testing-library/react'
import UserList from './UserList'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

afterEach(cleanup)

let usersArr

beforeEach(() => {
  usersArr = [
    { id: '1', profileImage: 'testImage', name: 'Anna' },
    { id: '2', profileImage: 'testImage', name: 'Alan' }
  ]
})

test('if renders', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <MemoryRouter>
      <UserList usersArr={usersArr} />
    </MemoryRouter>,
    div
  )
})

test('if matches snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <UserList usersArr={usersArr} />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
