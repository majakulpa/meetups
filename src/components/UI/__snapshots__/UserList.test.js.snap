// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
Array [
  <span
    className="font-bold text-lg"
>
     (
    2
    )
</span>,
  <div
    className="flex flex-wrap justify-center lg:justify-start"
>
    <a
        className="flex flex-col items-center w-32 md:w-48 bg-gray-100 rounded  p-2 md:p-3 m-1 md:m-3 hover:bg-gray-200 hover:shadow"
        href="/users/1"
        onClick={[Function]}
    >
        <div
            className="h-16 w-16 bg-cover rounded-full bg-center"
            style={
                Object {
                    "backgroundImage": "url(testImage})",
                  }
            }
            title="Profile Image"
        />
        <div
            className="text-sm font-medium m-3"
        >
            Anna
        </div>
    </a>
    <a
        className="flex flex-col items-center w-32 md:w-48 bg-gray-100 rounded  p-2 md:p-3 m-1 md:m-3 hover:bg-gray-200 hover:shadow"
        href="/users/2"
        onClick={[Function]}
    >
        <div
            className="h-16 w-16 bg-cover rounded-full bg-center"
            style={
                Object {
                    "backgroundImage": "url(testImage})",
                  }
            }
            title="Profile Image"
        />
        <div
            className="text-sm font-medium m-3"
        >
            Alan
        </div>
    </a>
</div>,
]
`;
